### ISSUE TITLE
"eth0" appears (with quotes, sometimes alone on its own line) in /etc/iiab/iiab.ini

### ISSUE THREAD

**Issue by holta** [iiab/iiab#1965]
This occurred on a fresh install being worked on 1 month ago in Mexico ("eth0" appeared in /etc/iiab/iiab.ini, including the quotation marks).  Concerning that the double-quotes appear (and should not).

Even more concerning is "eth0" that appeared twice in /etc/iiab/iiab.ini in South Africa on this IIAB machine built in April 2019 that was reinstalled in early September 2019: (again with double-quotes, but even more seriously when there is not variable name on the left, on the 4th line below)

```
...
[detected_network]
has_ifcfg_gw = none
prior_gateway_(device_gw2) = "eth0"
"eth0"
dhcpcd_result = enabled
network_manager_active = False
systemd_networkd_active = False
wan_in_interfaces = False
wireless_list_1(wifi1) = not found-1
wireless_list_2(wifi2) = wlan0
num_wifi_interfaces = 1
discovered_wireless_iface = wlan0
discovered_wired_iface = none
num_lan_interfaces = 1
gui_static_wan = False
iiab_lan_iface = br0
iiab_wan_iface = eth0
...
```

As excerpted from http://paste.debian.net/1102439/ which shows /etc/iiab/iiab.env as follows:

```
# This is a configuration file for IIAB
# It can be sourced in a shell script or read into an application
IIAB_BASE_PATH=/opt/iiab
IIAB_DIR=/opt/iiab/iiab
IIAB_RELEASE=7.0
IIAB_REVISION=0
IIAB_GATEWAY_ENABLED=True
LAN_IP=172.18.96.1
OS=raspbian
OS_VER=raspbian-9
WWWROOT=/library/www/html
STAGE=9
FQDN=box.lan
IIAB_USB_LIB_SHOW_ALL=True
IIAB_WAN_DEVICE="eth0"
IIAB_LAN_DEVICE="br0"
HOSTAPD_ENABLED=True
HTTPCACHE_ON=True
```

This results in Admin Console failing to install with the "eth0" error shown at bottom here:

> 2019-09-09 16:04:23,796 p=6070 u=root |  fatal: [127.0.0.1]: FAILED! => {"changed": true, "cmd": "/opt/admin/cmdsrv/scripts/iiab_update_menus.py", "delta": "0:00:00.340946", "end": "2019-09-09 16:04:23.714489", "msg": "non-zero return code", "rc": 1, "start": "2019-09-09 16:04:23.373543", "stderr": "Traceback (most recent call last):\n  File \"/opt/admin/cmdsrv/scripts/iiab_update_menus.py\", line 285, in <module>\n    main()\n  File \"/opt/admin/cmdsrv/scripts/iiab_update_menus.py\", line 73, in main\n    put_kiwix_enabled_into_menu_json()\n  File \"/opt/admin/cmdsrv/scripts/iiab_update_menus.py\", line 121, in put_kiwix_enabled_into_menu_json\n    kiwix.init()\n  File \"/opt/admin/cmdsrv/scripts/iiab_make_kiwix_lib.py\", line 195, in init\n    config.read(iiab_ini_file)\n  File \"/usr/lib/python2.7/ConfigParser.py\", line 305, in read\n    self._read(fp, filename)\n  File \"/usr/lib/python2.7/ConfigParser.py\", line 546, in _read\n    raise e\nConfigParser.ParsingError: File contains parsing errors: /etc/iiab/iiab.ini\n\t[line 107]: '\"eth0\"\\n'", "stderr_lines": ["Traceback (most recent call last):", "  File \"/opt/admin/cmdsrv/scripts/iiab_update_menus.py\", line 285, in <module>", "    main()", "  File \"/opt/admin/cmdsrv/scripts/iiab_update_menus.py\", line 73, in main", "    put_kiwix_enabled_into_menu_json()", "  File \"/opt/admin/cmdsrv/scripts/iiab_update_menus.py\", line 121, in put_kiwix_enabled_into_menu_json", "    kiwix.init()", "  File \"/opt/admin/cmdsrv/scripts/iiab_make_kiwix_lib.py\", line 195, in init", "    config.read(iiab_ini_file)", "  File \"/usr/lib/python2.7/ConfigParser.py\", line 305, in read", "    self._read(fp, filename)", "  File \"/usr/lib/python2.7/ConfigParser.py\", line 546, in _read", "    raise e", "ConfigParser.ParsingError: File contains parsing errors: /etc/iiab/iiab.ini", "\t[line 107]: '\"eth0\"\\n'"], "stdout": "Updating kiwix menus", "stdout_lines": ["Updating kiwix menus"]}
2019-09-09 16:04:23,803 p=6070 u=root |  PLAY RECAP ***************************************************************************************************************************************************
2019-09-09 16:04:23,804 p=6070 u=root |  127.0.0.1                  : ok=58   changed=24   unreachable=0    failed=1    skipped=8    rescued=0    ignored=0

Thanks @jvonau if you can help us with a strategy to find where this is likely arising from?

**Comment by holta** [comment]
FYI the South African IIAB machine's context @ http://paste.debian.net/1102439/ will delete itself in ~5 days on 2019-10-02 (LMK if a more permanent copy should be made!)

**Comment by holta** [comment]
> http://paste.debian.net/1102439/ will delete itself in ~5 days on 2019-10-02

These 1520 lines (South African iiab-diagnostics) are now re-posted for the coming year here:
https://paste.ubuntu.com/p/KJrygT7rNb/

**Comment by holta** [comment]
Can't replicate the line being split.

But appears largely resolved by PR #2278
